src/wurf/waf_configuration_context.py

```diff
@@ -3,18 +3,19 @@
 
 from waflib.Configure import ConfigurationContext
 
-from . import waf_resolve_context
 
-
-class WurfConfigurationContext(ConfigurationContext):
+class WafConfigurationContext(ConfigurationContext):
 
 
     def __init__(self, **kw):
-        super(WurfConfigurationContext, self).__init__(**kw)
+        super(WafConfigurationContext, self).__init__(**kw)
 
-    def execute(self):
+    def pre_recurse(self, node):
 
-        super(WurfConfigurationContext, self).execute()
+        super(WafConfigurationContext, self).pre_recurse(node)
 
-        # Call configure in all dependencies
-        waf_resolve_context.recurse_dependencies(self)
+        # Call configure() in all dependencies before executing configure()
```

Comment:
I think we should not make this tied to our waf-tools .. it if can be avoided.. therefore I would like to try to fix the waf-tools before trying to fix stuff here.